### PR TITLE
Users/hakohli/1303 remove invocation attachments

### PR DESCRIPTION
**Pull request by harleenkohli**
proposed fix for bug id #1303

### CODE REVIEW

File: src/Sarif/NotYetAutoGenerated/SarifRewritingVisitor.cs [hunk 1]
```diff
@@ -416,14 +416,6 @@ public virtual Invocation VisitInvocation(Invocation node)
                     }
                 }
 
-                if (node.Attachments != null)
```
Comment (harleenkohli) [inline src/Sarif/NotYetAutoGenerated/SarifRewritingVisitor.cs]: finally learnt about NotYetGenerated (the hard way :/)

Comment (michaelcfanning) [reply src/Sarif/NotYetAutoGenerated/SarifRewritingVisitor.cs]: nice!

---
In reply to: [259196449](https://github.com/Microsoft/sarif-sdk/pull/1305#discussion_r259196449) [](ancestors = 259196449)

File: src/ReleaseHistory.md [hunk 2]
```diff
@@ -229,7 +229,7 @@
 ## **v2.0.0-csd.2.beta.2019-01-09** [Sdk](https://www.nuget.org/packages/Sarif.Sdk/2.0.0-csd.2.beta.2019-01-09) | [Driver](https://www.nuget.org/packages/Sarif.Driver/2.0.0-csd.2.beta.2019-01-09) | [Converters](https://www.nuget.org/packages/Sarif.Converters/2.0.0-csd.2.beta.2019-01-09) | [Multitool](https://www.nuget.org/packages/Sarif.Multitool/2.0.0-csd.2.beta.2019-01-09)
 * BUGFIX: Result matching improvements in properties persistence.
 * FEATURE: Fortify FPR converter improvements.
-* API Non-BREAKING: Remove uniqueness requirement from 'result.locations'.
```
Comment (lgolding) [inline src/ReleaseHistory.md]: >Non [](start = 6, length = 3)

Thank you :-)

File: src/Sarif/NotYetAutoGenerated/SarifRewritingVisitor.cs (hunk 1 above)
Comment (lgolding) [reply src/Sarif/NotYetAutoGenerated/SarifRewritingVisitor.cs]: +1

---
In reply to: [259459661](https://github.com/Microsoft/sarif-sdk/pull/1305#discussion_r259459661) [](ancestors = 259459661,259196449)

File: src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs [hunk 3]
```diff
@@ -43,6 +43,12 @@ public static class PrereleaseCompatibilityTransformer
 
             switch (version)
             {
+                case "2.0.0-csd.2.beta.2019-02-20":
+                {
+                    // SARIF TC31. Nothing to do.
```
Comment (lgolding) [inline src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs]: >TC31 [](start = 29, length = 4)

"TC 31" -> "TC 32"

File: src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs [hunk 4]
```diff
@@ -43,6 +43,12 @@ public static class PrereleaseCompatibilityTransformer
...
+                    break;
+                }
+
                 case "2.0.0-csd.2.beta.2019-01-24":
                 {
                     // SARIF TC31. Nothing to do.
```
Comment (lgolding) [inline src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs]: >Nothing to do [](start = 35, length = 13)

There _is_ something to do between TC 31 and TC 32: You have to move any `invocation.attachments` and `tool.extensions`. The properties between an `attachment` and a `toolComponent` (the array element of `tool.extensions`) don't completely line up, but they do at least both have file locations.

@mikefan@microsoft.com If you think nobody has used `invocation.attachments` and it's not worth doing this, that's fine, but it deserves at least a comment.

File: src/Sarif.UnitTests/TestData/InsertOptionalDataVisitor/ExpectedOutputs/CoreTests_Hashes+TextFiles+ComprehensiveRegionProperties+RegionSnippets+ContextRegionSnippets+FlattenedMessages.sarif [hunk 5]
```diff
@@ -119,7 +119,7 @@
                   "startColumn": 1,
                   "endLine": 2,
                   "endColumn": 14,
-                  "charOffset":  0,
```
Comment (lgolding) [inline src/Sarif.UnitTests/TestData/InsertOptionalDataVisitor/ExpectedOutputs/CoreTests_Hashes+TextFiles+ComprehensiveRegionProperties+RegionSnippets+ContextRegionSnippets+FlattenedMessages.sarif]: >   [](start = 31, length = 2)

I love it :-)

**Pull request merged**